addons/sale_loyalty/models/sale_order.py

```diff
@@ -582,6 +582,21 @@ def _write_vals_from_reward_vals(self, reward_vals, old_lines, delete=True):
         self.write({'order_line': command_list})
         return self.env['sale.order.line'] if delete else old_lines[len(reward_vals):]
 
+
+    def _is_better_global_discount_applied(self, current_reward, new_reward):
+        assert current_reward
+
+        def get_max_discount(reward):
```

Comment:
This method returns the discount applicable based on the reward and the sale order. It is based on the `self._discountable_order()` and does not return the max discount but the discount that can be applied on the sale order.
```suggestion
        def compute_discount(reward):
```